hydra-cluster/src/Hydra/Cluster/Faucet.hs

```diff
@@ -148,88 +143,85 @@ seedFromFaucetBlockfrost receivingVerificationKey lovelace = do
+  void $ returnFundsToFaucet' tracer backend (snd senderKeys)
 
 returnFundsToFaucet' ::
+  ChainBackend backend =>
   Tracer IO FaucetLog ->
-  RunningNode ->
+  backend ->
   SigningKey PaymentKey ->
   IO Coin
-returnFundsToFaucet' tracer RunningNode{networkId, nodeSocket} senderSk = do
+returnFundsToFaucet' tracer backend senderSk = do
   (faucetVk, _) <- keysFor Faucet
+  networkId <- Backend.queryNetworkId backend
   let faucetAddress = mkVkAddress networkId faucetVk
   let senderVk = getVerificationKey senderSk
-  utxo <- queryUTxOFor networkId nodeSocket QueryTip senderVk
+  utxo <- Backend.queryUTxOFor backend QueryTip senderVk
   returnAmount <-
     if null utxo
       then pure 0
       else retryOnExceptions tracer $ do
         let utxoValue = balance @Tx utxo
         let allLovelace = selectLovelace utxoValue
         tx <- sign senderSk <$> buildTxBody utxo faucetAddress
-        submitTransaction networkId nodeSocket tx
-        void $ awaitTransaction networkId nodeSocket tx
+        Backend.submitTransaction backend tx
+        void $ Backend.awaitTransaction backend tx faucetVk
         pure allLovelace
   traceWith tracer $ ReturnedFunds{returnAmount}
   pure returnAmount
  where
   buildTxBody utxo faucetAddress =
     -- Here we specify no outputs in the transaction so that a change output with the
     -- entire value is created and paid to the faucet address.
-    buildTransaction networkId nodeSocket faucetAddress utxo [] [] >>= \case
+    buildTransaction backend faucetAddress utxo [] [] >>= \case
       Left e -> throwIO $ FaucetFailedToBuildTx{reason = e}
       Right tx -> pure $ getTxBody tx
 
 -- Use the Faucet utxo to create the output at specified address
 createOutputAtAddress ::
-  RunningNode ->
+  ChainBackend backend =>
+  NetworkId ->
+  backend ->
   AddressInEra ->
   TxOutDatum CtxTx ->
   Value ->
   IO (TxIn, TxOut CtxUTxO)
-createOutputAtAddress node@RunningNode{networkId, nodeSocket} atAddress datum val = do
+createOutputAtAddress networkId backend atAddress datum val = do
   (faucetVk, faucetSk) <- keysFor Faucet
-  utxo <- findFaucetUTxO node 0
+  utxo <- findFaucetUTxO networkId backend 0
   let collateralTxIns = mempty
   let output = TxOut atAddress val datum ReferenceScriptNone
-  buildTransaction
-    networkId
-    nodeSocket
-    (changeAddress faucetVk)
-    utxo
-    collateralTxIns
-    [output]
-    >>= \case
-      Left e ->
-        throwErrorAsException e
-      Right x -> do
-        let body = getTxBody x
-        let tx = makeSignedTransaction [makeShelleyKeyWitness body (WitnessPaymentKey faucetSk)] body
-        submitTransaction networkId nodeSocket tx
-        newUtxo <- awaitTransaction networkId nodeSocket tx
-        case UTxO.find (\out -> txOutAddress out == atAddress) newUtxo of
-          Nothing -> failure $ "Could not find script output: " <> decodeUtf8 (encodePretty newUtxo)
-          Just u -> pure u
- where
-  changeAddress = mkVkAddress networkId
```

Comment:
I think having this variable makes the code more readable.

Reply:
How? It is used at only one place?